tests/sentry/models/tests.py

```diff
@@ -91,72 +91,45 @@ class EventNodeStoreTest(TestCase):
     def test_event_node_id(self):
         # Create an event without specifying node_id. A node_id should be generated
         e1 = Event(project_id=1, event_id="abc", data={"foo": "bar"})
-        e1.save()
-        e1_node_id = e1.data.id
         assert e1.data.id is not None, "We should have generated a node_id for this event"
-        e1_body = nodestore.get(e1_node_id)
+        e1_node_id = e1.data.id
         e1.data.save()
         e1_body = nodestore.get(e1_node_id)
         assert e1_body == {"foo": "bar"}, "The event body should be in nodestore"
 
-        e1 = Event.objects.get(project_id=1, event_id="abc")
+        e1 = Event(project_id=1, event_id="abc")
+
         assert e1.data.data == {"foo": "bar"}, "The event body should be loaded from nodestore"
         assert e1.data.id == e1_node_id, "The event's node_id should be the same after load"
 
-        # Create another event that references the same nodestore object as the first event.
-        e2 = Event(project_id=1, event_id="def", data={"node_id": e1_node_id})
-        assert e2.data.id == e1_node_id, "The event should use the provided node_id"
-        e2_body = nodestore.get(e1_node_id)
-        assert e2_body == {"foo": "bar"}, "The event body should be in nodestore already"
-        e2.save()
-        e2_body = nodestore.get(e1_node_id)
-        assert e2_body == {"foo": "bar"}, "The event body should not be overwritten by save"
-
-        e2 = Event.objects.get(project_id=1, event_id="def")
-        assert e2.data.data == {"foo": "bar"}, "The event body should be loaded from nodestore"
-        assert e2.data.id == e1_node_id, "The event's node_id should be the same after load"
-
-        # Create an event with a new event body that specifies the node_id to use.
-        e3 = Event(project_id=1, event_id="ghi", data={"baz": "quux", "node_id": "1:ghi"})
-        assert e3.data.id == "1:ghi", "Event should have the specified node_id"
-        assert e3.data.data == {
-            "baz": "quux"
-        }, "Event body should be the one provided (sans node_id)"
-        e3.save()
-        e3_body = nodestore.get("1:ghi")
-        e3.data.save()
-        e3_body = nodestore.get("1:ghi")
-        assert e3_body == {"baz": "quux"}, "Event body should be saved to nodestore"
-
-        e3 = Event.objects.get(project_id=1, event_id="ghi")
-        assert e3.data.data == {"baz": "quux"}, "Event body should be loaded from nodestore"
-        assert e3.data.id == "1:ghi", "Loaded event should have the correct node_id"
-
-        # Try load it again, but using the pickled/compressed string we would expect to find
-        # in the column
-        e3_pickled_id = compress(pickle.dumps({"node_id": "1:ghi"}))
-        e3 = Event(project_id=1, event_id="jkl", data=e3_pickled_id)
-        assert e3.data.data == {"baz": "quux"}, "Event body should be loaded from nodestore"
```

Comment:
One last thing: this was testing that an event could be created specifying the node id. Are you confident this behavior did not exist in production ?

Reply:
Well, we are ultimately still specifying a generated node id to be saved, but this mechanism that performs a nodestore load from a data object with a node_id is specific to the way NodeField works and we are not using that anymore.

Reply:
ok